libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRouterRefreshCallback.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.base.route
+
+interface NavigationRouterRefreshCallback {
+    fun onRefreshReady(route: NavigationRoute)
+    fun onFailure(error: NavigationRouterRefreshError)
+}
+
+class NavigationRouterRefreshError internal constructor(
+    val message: String? = null,
+    val throwable: Throwable? = null,
+    val routerFailure: RouterFailure? = null,
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is NavigationRouterRefreshError) return false
```

Comment:
missin `inheritors` I think, should be 
```suggestion
       if (javaClass != other?.javaClass) return false
```